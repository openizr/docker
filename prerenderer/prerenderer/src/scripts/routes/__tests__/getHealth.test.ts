/**
 * Copyright (c) Matthieu Jabbour. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import getHealth from 'scripts/routes/getHealth';
import { FastifyRequest, FastifyReply } from 'fastify';

jest.mock('fastify');

describe('routes/getHealth', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('correctly handles request', () => {
    const send = jest.fn();
    getHealth.handler({} as FastifyRequest, { send } as unknown as FastifyReply);
    expect(getHealth.schema).toMatchSnapshot();
    expect(send).toHaveBeenCalledTimes(1);
    expect(send).toHaveBeenCalledWith();
  });
});
