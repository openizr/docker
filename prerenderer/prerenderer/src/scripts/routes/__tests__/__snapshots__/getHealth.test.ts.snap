// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes/getHealth correctly handles request 1`] = `
Object {
  "response": Object {
    "200": Object {
      "type": "string",
    },
  },
}
`;
